File: RobotSimulatorC#/RobotSimulatorTestApp/RobotSimulator.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace RobotSimulatorTestApp
{
    public class RobotSimulator
    {
        private IMove? _selectedMove;
        private IRotation? _rotation;

        public RobotSimulator() 
        {
            _selectedMove = null;
            _rotation = null;
        }

        // implemented dependency injection in both Move and SelectedRotation method

        public void Move(DataBlock block)
        {
            if (block.CurrentDirection == Direction.N)
            {               
                _selectedMove = new MoveUp();

                block.CurrentRow = _selectedMove.Move(block.CurrentRow, 0);
            }
            else if (block.CurrentDirection == Direction.E)
            {
                _selectedMove = new MoveRight();
                block.CurrentColumn = _selectedMove.Move(block.CurrentColumn, block.Columns);
            }
            else if (block.CurrentDirection == Direction.S)
            {
                _selectedMove = new MoveDown();
                block.CurrentRow = _selectedMove.Move(block.CurrentRow, block.Rows);
            }
            else if (block.CurrentDirection == Direction.W)
            {
                _selectedMove = new MoveLeft();
                block.CurrentColumn = _selectedMove.Move(block.CurrentColumn, 0);
            }
        }
                
        public Direction SelectedRotation(IRotation rotation, Direction currentDirection)
        {
            _rotation = rotation;
            return _rotation.Rotate(currentDirection);
        }
    }
}

File: RobotSimulatorJava/RobotSimulator.java
public class RobotSimulator {
    private IMove selectedMove;
    private IRotation rotation;

    public RobotSimulator() {
        this.selectedMove = null;
        this.rotation = null;
    }

    // Dependency injection implemented in both move and selectedRotation methods
    public void move(DataBlock block) {
        if (block.getCurrentDirection() == Direction.N) {
            selectedMove = new MoveUp();
            block.setCurrentRow(selectedMove.move(block.getCurrentRow(), 0));
        } 
        else if (block.getCurrentDirection() == Direction.E) {
            selectedMove = new MoveRight();
            block.setCurrentColumn(selectedMove.move(block.getCurrentColumn(), block.getColumns()));
        } 
        else if (block.getCurrentDirection() == Direction.S) {
            selectedMove = new MoveDown();
            block.setCurrentRow(selectedMove.move(block.getCurrentRow(), block.getRows()));
        } 
        else if (block.getCurrentDirection() == Direction.W) {
            selectedMove = new MoveLeft();
            block.setCurrentColumn(selectedMove.move(block.getCurrentColumn(), 0));
        }
    }

    public Direction selectedRotation(IRotation rotation, Direction currentDirection) {
        this.rotation = rotation;
        return this.rotation.rotate(currentDirection);
    }
}
